engine/src/main/java/org/pentaho/di/repository/RepositoryConnectionUtils.java

```diff
@@ -0,0 +1,179 @@
+/*!
+ * This program is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License, version 2.1 as published by the Free Software
+ * Foundation.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along with this
+ * program; if not, you can obtain a copy at http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+ * or from the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ *
+ * Copyright (c) 2019 Hitachi Vantara.  All rights reserved.
+ */
+
+package org.pentaho.di.repository;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.pentaho.di.core.exception.KettleException;
+import org.pentaho.di.core.exception.KettleSecurityException;
+import org.pentaho.di.core.logging.LoggingBuffer;
+import org.pentaho.di.core.plugins.PluginRegistry;
+import org.pentaho.di.core.plugins.RepositoryPluginType;
+import org.pentaho.di.i18n.BaseMessages;
+import org.pentaho.di.repository.RepositoriesMeta;
+import org.pentaho.di.repository.RepositoryMeta;
+import org.pentaho.platform.api.engine.ActionExecutionException;
+import org.pentaho.platform.api.engine.IAuthorizationPolicy;
+import org.pentaho.di.repository.Repository;
+
+import java.io.ByteArrayInputStream;
+
+/**
+ *  Provides a utility method to connect to repository and return the repository object
+ */
+public class RepositoryConnectionUtils {
+  private static final Log logger = LogFactory.getLog( RepositoryConnectionUtils.class );
+  private static final String SINGLE_DI_SERVER_INSTANCE = "singleDiServerInstance";
+
+
+  /**
+   * Connects to the PDI repository. If the  system settings "SINGLE_DI_SERVER_INSTANCE" is set, it will connect to
+   * repository using the embedded repository.xml otherwise it will connect to repository with the given name.
+   *
+   * @param  String RepositoryName
+   * @param LoggingBuffer
+   * @return  Repository
+   * @throws KettleException
+   * @throws KettleSecurityException
+   * @throws ActionExecutionException
+   */
+  public static Repository connectToRepository( final String repositoryName, boolean isSingleDiServerInstance,
+             IAuthorizationPolicy authorizationPolicy, String userName,
+             String fullyQualifiedServerUrl, LoggingBuffer logBuffer ) throws KettleSecurityException, KettleException,
+              ActionExecutionException {
+
+    if ( !hasExecutePermission( authorizationPolicy ) ) {
+      throw new IllegalStateException(
+        BaseMessages.getString( "RepositoryConnectionUtils.ERROR_0001_NO_PERMISSION_TO_EXECUTE" ) ); //$NON-NLS-1$
+    }
```

Comment:
Alternatively, this util method - whose sole purpose is to establish a Repository connection - doesn't do a check for `hasExecutePermission( authorizationPolicy )` , as it does not pertain its realm of responsibilities.  

Other blocks of code which would leverage this `RepositoryConnectionUtils. connectToRepository()` would be the ones tasked with checking if a certain `username` has execute permissions or not.

Reply:
I thought we discussed it and decided to keep the check.